test/integration/statsDTests.coffee

```diff
@@ -0,0 +1,132 @@
+
+      before (done) ->
+        Statsd = require "../../lib/api/statsd"
+        Statsd.authenticated = auth.getAuthDetails()
+        Statsd.request = {}
+        Statsd.request.query = {}
+        done()
+
+
+      after (done) ->
+        done()
+
+
+      it "should fetch global load Time metrics",  (done) ->
+        mock = sinon.mock(Statsd)
+        mock.expects('fetchData').once().withExactArgs("/render?target=transformNull(summarize(stats.timers." + domain + ".channels.sum,'1hour','avg'))&from=-1days&format=json")
+        `co(function* () {
+          yield Statsd.retrieveAverageLoadTimePerHour();
+        })`
+        mock.verify()
+        mock.restore()
+        done()
+
+
+      it "should fetch global status metrics ",  (done) ->
+        mock = sinon.mock(Statsd);
+        mock.expects('fetchData').once().withExactArgs("/render?target=transformNull(summarize(stats.counters." + domain + ".channels.jjhreujiwh.statuses.Processing.count,'1day'))&format=json")
+        `co(function* () {
+           yield Statsd.fetcGlobalStatusMetrics(['jjhreujiwh']);
+        })`
+        mock.verify()
+        mock.restore()
+        done()
+
+
+      it "should fetch channel transaction count metrics ", (done) ->
+          mock = sinon.mock(Statsd);
+          mock.expects('fetchData').once().withExactArgs("/render?target=transformNull(summarize(stats.counters." + domain + ".channels.jjhreujiwh.count,'1day'))&from=-7days&format=json&target=transformNull(summarize(stats.timers." + domain + ".channels.jjhreujiwh.sum,'1day','avg'))")
+          `co(function* () {
+            yield Statsd.retrieveChannelMetrics('count','jjhreujiwh');
+          });`
+          mock.verify();
+          mock.restore();
+          done();
+
+
+      it "should fetch channel status metrics ",  (done) ->
+          mock = sinon.mock(Statsd)
+          mock.expects('fetchData').once().withExactArgs("/render?target=transformNull(summarize(stats.counters." + domain + ".channels.jjhreujiwh.statuses.Processing.count,'1week'))&format=json")
+          `co(function* () {
+             yield Statsd.retrieveChannelMetrics('status','jjhreujiwh');
+          });`
+          mock.verify()
+          mock.restore()
+          done()
+
+      it "should convert the result from statsd to the correct format", (done) ->
+        fetchDataObject = {}
+        fetchDataObject.data = [
+          [
+            83
+            1421280000
+          ]
+        ]
+
+        fetchDataObject.data1 = [
+          [
+            200
+            1421280000
+          ]
+        ]
+        convertToRequiredFormatSpy = sinon.spy Statsd, 'convertToRequiredFormat'
+        data = convertToRequiredFormatSpy fetchDataObject, 'retrieveAverageLoadTimePerHour'
+        data2 = convertToRequiredFormatSpy fetchDataObject, 'retrieveChannelMetrics'
+        data[0].avgResp.should.be.exactly 83
+        data2[0].avgResp.should.be.exactly 200
+        convertToRequiredFormatSpy.restore()
+        done()
+
+
```

Comment:
Here I still think we need more tests. You are testing that the correct `fetchData` call is made but there are no tests to check that when data is returned from the `fetchData` function and it is processed that it is returned in the format that you expect. Please could you add some tests to test this.

I'm sure with sinon that you can make a mock that returns certain predefined data, then you would just need to check that that data gets returned in the format that you expect.